src/app/body/wishlist/wishlist-page/wishlist-page.component.ts

```diff
@@ -0,0 +1,36 @@
+import { Component, OnDestroy, OnInit } from '@angular/core';
+import { Product } from '../../../models/product.model';
+import { WishlistService } from '../../../services/wishlist.service';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Component({
+  selector: 'app-wishlist-page',
+  templateUrl: './wishlist-page.component.html',
+  styleUrls: ['./wishlist-page.component.scss']
+})
+export class WishlistPageComponent implements OnInit, OnDestroy {
+  private destroy$ = new Subject();
+  public wishlist: Product[];
+
+  constructor(private wishlistService: WishlistService) { }
+
+  ngOnInit(): void {
+    this.wishlistService.getWishlist().then();
+    this.subscribeToWishlist();
```

Comment:
1. Why do you need the second method, if you can get value from 
```ts
this.wishlistService.getWishlist().then((wishlist: Product[]) => this.wishlist = wishlist);
```
can't you?

2. Why do you need `WishlistResolver` if you don't use it?

Reply:
1 Yes I can